### PR TITLE
feat: update jupyterlab and add extension for notebook style headings…

### PR DESCRIPTION
**Pull request by jeffmm**
# Context

This updates the `orbyter-ml-dev` to v3.5 which updates jupyterlab to v3.0.16 and adds the `collapsible_headings` jupyterlab extension and installs the `dask-cloudprovider` package.

Edit: added `dask-labextension==5.0.2`

# Changes

* Changes image version to 3.5
* Adds `aquirdturtle-collapsible-headings`, `dask-cloudprovider`, and `dask-labextension==5.0.2` packages
* Updates docs to reflect changes
* Package additions/updates:
  - aquirdturtle-collapsible-headings==3.1.0
  - black==21.5b2
  - boto3==1.17.92
  - dask[complete]==2021.6.0
  - dask-cloudprovider==2021.3.1
  - dask-labextension==5.0.2
  - dask-ml==1.9.0
  - dvc==2.3.0
  - flake8==3.9.2
  - ipdb==0.13.9
  - ipython==7.24.1
  - lightgbm==3.2.1
  - line-profiler==3.3.0
  - matplotlib==3.4.2
  - mlflow==1.17.0
  - more-itertools==8.8.0
  - mypy==0.902
  - notebook==6.4.0
  - numpy==1.20.3
  - pandas==1.2.4
  - pyarrow==4.0.1
  - pytest==6.2.4
  - python-dotenv==0.17.1
  - pytictoc==1.5.2
  - scikit-learn==0.24.2
  - scipy==1.6.3
  - shap==0.39.0
  - Sphinx==4.0.2
  - SQLAlchemy==1.4.18
  - streamlit==0.82.0
  - tqdm==4.61.0
  - xarray==0.18.2
  - xgboost==1.4.2

### CODE REVIEW

File: orbyter-ml-dev/README.md [hunk 1]
```diff
@@ -35,30 +120,30 @@ Pin `ipython==7.20.0` to avoid `jedi` autocompletion bug with earlier versions
 #### Python package updates
 
 ##### Updated
-* boto3==1.17.10
```
Comment (jeffmm) [inline orbyter-ml-dev/README.md:38]: Everything below here hasn't changed, it's just that my markdown linter changed all the bullets from asterisks to dashes.

File: orbyter-ml-dev/requirements.txt [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
+python-dotenv==0.17.1
+pytictoc==1.5.2
+scikit-learn==0.24.2
+scipy==1.6.3
 seaborn==0.11.1
-shap==0.38.1
-Sphinx==3.5.1
-SQLAlchemy==1.3.23
+shap==0.39.0
+Sphinx==4.0.2
```
Comment (mstefferson) [inline orbyter-ml-dev/requirements.txt:39]: Not sure how careful we need to about major version bumps

File: orbyter-ml-dev/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# orbyter-ml-dev:3.3
+# orbyter-ml-dev:3.5
```
Comment (mstefferson) [inline orbyter-ml-dev/Dockerfile:1]: what happened to `3.4`

**Review by mstefferson — APPROVED** [review]
LGTM

File: orbyter-ml-dev/requirements.txt (hunk 2 above)
Comment (jeffmm) [reply orbyter-ml-dev/requirements.txt:39]: Good question, and this one might be worth looking into, I'll make sure there isn't anything breaking in their changelog

File: orbyter-ml-dev/Dockerfile (hunk 3 above)
Comment (jeffmm) [reply orbyter-ml-dev/Dockerfile:1]: This comment never got updated. 3.4 does already exist, I just hadn't noticed the update.

**Comment by jeffmm** [comment]
Holding off on merging for the time being while I review adding the [Dask Jupyterlab extension](https://www.youtube.com/watch?v=EX_voquHdk0)

**Review by rameshvs — APPROVED** [review]
LGTM!

**Pull request merged**